tla/counterfeit-cycles/counterfeit_cycles.tla

```diff
@@ -0,0 +1,293 @@
+             \* Initially, all subnets behave honestly.
+             honest |-> TRUE
+         ]
+     ]
+  /\ subnetMsgs = <<>>
+  
+LedgerInit ==
+  \* Initialize the state of the ledger.
+  ledger = [
+    \* The balance of each subnet.
+    balances |-> [s \in SUBNETS |-> STARTING_BALANCE_PER_SUBNET],
+    
+    \* The queue of messages to be processed.
+    msgs |-> <<>>,
+    
+    \* The set of subnets that the ledger thinks is dishonest.
+    \* NOTE: I think this is redundant and is not needed for this model.
+    dishonestSubnets |-> {}
+  ]
+
+Init ==
+  /\ LedgerInit
+  /\ SubnetsInit
+
+-----------------------------------------------------------------------------
+(***************************************************************************)
+(* Type checks                                                             *)
+(***************************************************************************)
+
+TransferMessage ==
+  \* A message to transfer cycles between subnets.
+  [type : {"transfer"}, from : SUBNETS, to : SUBNETS, amount : Nat \ {0}]
+
+\* The set of all possible messages that are sent/received by subnets.
+SubnetMessages ==
+  \* Transferring cycles from one subnet to another.
+  TransferMessage
+    \union
+    
+  \* Approving a cycles transfer.
+  \* This message is sent by the ledger to indicate that the `amount` sent is
+  \* now finalized.
+  [type : {"approve"}, from : SUBNETS, to : SUBNETS, amount : Nat \ {0}]  
+
+LedgerMessages ==
+  \* The set of all possible messages that can be received by the Ledger.
+  TransferMessage
+
+SubnetsTypeOK ==
+  /\ \A i \in DOMAIN subnetMsgs: {subnetMsgs[i]} \subseteq SubnetMessages
+  /\ subnets \in [SUBNETS -> [
+        balance: Nat,
+        unfinalized: [SUBNETS -> Nat],
+        msgsToLedger: Seq(LedgerMessages),
+        honest: {TRUE, FALSE}
+     ]]
+
+LedgerTypesOK ==
+  \* Type-correctness invariant.
+  /\ DOMAIN ledger.balances = SUBNETS
+  /\ \A s \in SUBNETS: ledger.balances[s] >= 0
+  /\ ledger.dishonestSubnets \subseteq SUBNETS
+  /\ \A i \in DOMAIN ledger.msgs: {ledger.msgs[i]} \subseteq LedgerMessages
+  
+TypesOK ==
+  \* Type-correctness invariant.
+  /\ LedgerTypesOK
+  /\ SubnetsTypeOK
+
+-----------------------------------------------------------------------------
+(***************************************************************************)
+(* Subnet actions                                                          *)
+(***************************************************************************)
+
+SubnetSendTransfer ==
+  (* Honestly sends a transfers cycles from one subnet to another. *)
+  
+  \* Choose two subnets.
+  \E sender, receiver \in SUBNETS: sender /= receiver
+  /\ subnets[sender].balance > 0
+  
+  \* Choose an amount that's within the subnet's balance.
+  /\ \E amount \in 1..subnets[sender].balance:
+  
+  \* Send a transfer to the receiver.
+  /\ subnetMsgs' = Append(subnetMsgs, [
+        type |-> "transfer",
```

Comment:
One trick I sometimes do is to define constants for these things, like `TRANSFER == "transfer"`. That way the parser at least complains about an unknown identifier if I mistype things. 
Just to mention, there's also another analysis tool for TLC:
https://apalache.informal.systems/
that comes with a type checker for TLA (and since recently also some kind of typed frontend language, but I haven't tried it yet).

Reply:
A style nitpick: I normally indent the clauses under a quantifier (i.e., everything below `\E sender...` in this definition) so I can see where they are bound, but that's obviously personal preference.